pkg/cri/resource-manager/policy/builtin/memtier/node.go

```diff
@@ -403,6 +407,83 @@ func (n *node) GetMemoryType() memoryType {
 	return memoryMask
 }
 
+// MoveAllWorkloadsUp transferes all containers from this node to its
+// parent node.
+func (n *node) MoveAllWorkloadsUp() (bool, error) {
+
+	currentSupply := n.FreeSupply()
+	parentSupply := n.parent.FreeSupply()
+	changed := false
+	p := n.policy
+
+	// Do not change the map while iterating through it.
+	newAllocations := make(map[string]Grant)
+
+	for cid, grant := range p.allocations.grants {
+		if grant.GetNode().IsSameNode(n) {
+			log.Debug("Moving container %s up to node %s", cid, n.parent)
+			req := grant.Request()
+			currentSupply.Release(grant)
+			newGrant, err := parentSupply.Allocate(req)
```

Comment:
I think we should not try doing it this way. If we do it like this, then we genuinely reallocate the container which will result in a reallocation of its cpusets as well. Instead we should only 'relax'/'elevate'/'broaden' the cpuset.mems pinning/allocation up in the tree.